core/src/avm2/globals/flash/xml/XMLNode.as

```diff
@@ -1,112 +1,193 @@
 package flash.xml
 {
-   import __ruffle__.stub_method;
-   import __ruffle__.stub_setter;
-
-   // TODO: Re-implement XMLNode shim in Rust.
-   public class XMLNode {
-      internal var _isDocument: Boolean = false;
-      internal var _xml: XML;
-      internal var _parent: XMLNode = null;
-      internal var _children: Array = null;
-      internal var _index: uint = 0;
-
-      public function XMLNode(type: uint, input: String) {
-         _setXML(input);
-      }
-
-      internal function _setXML(input: String, ignoreWhite: Boolean = false) {
-         var previousSettings = XML.AS3::settings();
-         try {
-            XML.ignoreWhitespace = ignoreWhite;
-            _xml = new XML(input);
-         } finally {
-            XML.AS3::setSettings(previousSettings);
-         }
-      }
-
-      public function get attributes(): Object {
-         var attrs = new Object();
-         for each (var attribute in _xml.attributes()) {
-            attrs[attribute.localName()] = attribute.toString();
-         }
-         return attrs;
-      }
-
-      public function get childNodes(): Array {
-         // Cache the children to avoid creating new XMLNodes on
-         // every access.
-         if (_children) {
-            return _children;
-         }
-
-         var children = _xml.children();
-         var nodes = new Array();
-         for (var i = 0; i < children.length(); i++) {
-            var node = new XMLNode(0, children[i].toXMLString());
-            node._parent = this;
-            node._index = i;
-            nodes.push(node);
-         }
-         _children = nodes;
-         return nodes;
-      }
-
-      public function get firstChild(): XMLNode {
-         return childNodes[0];
-      }
-
-      public function get nextSibling(): XMLNode {
-         return this._parent.childNodes[this._index + 1];
-      }
-
-      public function get nodeName(): String {
-         if (_isDocument) {
+
+import __ruffle__.stub_getter;
```

Comment:
nit: fmt (indent everything from here to end of file)

Reply:
Indented it all.